beater/v2_handler.go

```diff
@@ -108,8 +116,22 @@ func (v *v2Handler) Handle(beaterConfig *Config, report publish.Reporter) http.H
 			v.sendResponse(logger, w, &sr)
 			return
 		}
+		ctx := r.Context()
+
+		// fetch the rate limiter from the cache, if a cache is given
+		var limiter *rate.Limiter
+		if v.cache != nil && beaterConfig.RumConfig.EventRate != nil {
+			key := utility.RemoteAddr(r)
+			if !v.cache.Contains(key) {
+				v.cache.Add(key, rate.NewLimiter(rate.Limit(beaterConfig.RumConfig.EventRate.Limit),
+					beaterConfig.RumConfig.EventRate.Limit*2))
```

Comment:
`2` should be a constant. I also think 2 is [perhaps not right](https://github.com/elastic/apm-server/pull/1367#issuecomment-420235843) for RUM. WDYT?

Reply:
As far as I understand `event_rate.limit:1000` means a single IP can send 1000 events per second (= 2 page loads per second) continuously and burst up to 5000 (= 10 page loads per second). 500 spans is the max that backend agents allow by default. The avg. number of spans is much lower. The 2 page loads per second is then more like a minimum, because most transactions will have much less than 500 spans. That seems excessive to me.

Is it correct that if the burst limit is reached, our rate limiter would reject the request instead of delaying the reading?

Reply:
A request is rejected if the burst limit is reached and the rate limit would still be exceeded after a defined waiting time (1 second). 
This means that if only one request is sent, that exceeds the limit, the first `n=limit` events are processed, and then the processing is throttled. One event after the other is checked, and if the limit is already reached, the event is added to the burst queue. If the burst queue is full and the waiting time until this event could be accepted is larger than a defined timeout (1 second) then the processing fails with an ip rate limiting error. 
If several requests from one IP are sent concurrently then `limit + burst` requests can be processed in a throttled way. If more requests are sent then they are either rejected or some events get processed in a throttled way and then they get rejected. 

Having a large burst size makes sense if we think that there won't be a lot of concurrent requests and some of the requests will be exceptionally large, so we are fine with processing them slowly.